client/src/components/auth/userform.js

```diff
@@ -78,15 +78,44 @@ const UserForm = props => {
           props.onSubmit();
         }}
       >
-        {({ errors, touched }) => {
-          let validity = !(
-            !!errors.username ||
-            !!errors.password ||
-            !!errors.email
-          );
-          if (validity !== props.valid) {
-            props.errorComm(validity);
+        {({ errors, touched, isValid, handleChange }) => {
+          let validity = {
+            username: !!errors.username,
+            password: !!errors.password,
+            email: !!errors.email
+          };
+
+          console.log(errors);
+          console.log(touched);
+          console.log(props.handleTouch);
+          console.log(props.email_touched);
+          console.log(validity);
+          console.log(props.valid);
+          if (
+            !!touched.email !== !!props.email_touched &&
+            !props.email_touched
+          ) {
+            props.handleTouch("email_touched");
+          }
```

Comment:
if either touched.email is not the same value as props.email_touched (the saved state of email being touched, we should update the saved one. That is the stuff before the and. We only ever want to change it from false to true so we don't want to make any changes if it's already on true, so the part after the and makes sure we don't update the saved state to be false if it's already true. So changes only happen if it starts on false.

Reply:
Translation provided by Zach ^